File: FE/src/app/_models/userReservation.ts
import { MovieSeat } from "./movieSeat";
import { ParkingPlace } from "./parkingPlace";
import { Movie } from "./movie";

export class UserReservation {
    UserId: Number;
    ReservedSeats: MovieSeat[];
    ReservedParkingPlaces: ParkingPlace[];
    ReservedMovies: Movie[];
}

File: API/Models/UserReservation.cs
using System.Collections.Generic;

namespace API.Models
{
    public class UserReservation
    {
        public long UserId { get; set; }
        public List<Movie> ReservedMovies { get; set; }
        public List<MovieSeat> ReservedMovieSeats { get; set; }
        public List<ParkingPlace> ParkingPlaces { get; set; }
    }
}